### PR TITLE
Add php8 support

### LINKED ISSUE #8: PHP 8.0 support
### Feature Request

|    Q        |   A
|------------ | ------
| New Feature | yes

#### Summary

To be prepared for the december release of PHP 8.0, this repository has some additional TODOs to be tested against the new major version.

In order to make this repository compatible, one has to follow these steps:

- [ ] Modify `composer.json` to provide support for PHP 8.0 by adding the constraint `~8.0.0`
- [ ] Modify `composer.json` to drop support for PHP less than 7.3
- [ ] Modify `composer.json` to implement phpunit 9.3 which supports PHP 7.3+
- [ ] Modify `.travis.yml` to ignore platform requirements when installing composer dependencies (simply add `--ignore-platform-reqs` to `COMPOSER_ARGS` env variable)
- [ ] Modify `.travis.yml` to add PHP 8.0 to the matrix (*NOTE*: Do not allow failures as PHP 8.0 has a feature freeze since 2020-08-04!)
- [ ] Modify source code in case there are incompatibilities with PHP 8.0

### PR DESCRIPTION
**Pull request by marcelthole**
|    Q          |   A
|-------------- | ------
| Documentation | no
| Bugfix        | no
| BC Break      | no
| New Feature   | no
| RFC           | no
| QA            | yes

### Description

fixes #8

### CODE REVIEW

**Comment by marcelthole** [comment]
The PHP8 Build failed on travis, because `fig/http-message-util` doesn't allow PHP 8. See https://github.com/php-fig/http-message-util/pull/19

File: test/Middleware/RouteMiddlewareFactoryTest.php [hunk 1]
```diff
@@ -63,7 +65,11 @@ public function testFactoryAllowsSpecifyingRouterServiceViaConstructor()
         $middleware = $factory($this->container->reveal());
 
         $this->assertInstanceOf(RouteMiddleware::class, $middleware);
-        $this->assertAttributeSame($router, 'router', $middleware);
+
+        $middlewareRouter = Closure::bind(function () {
```
Comment (marcelthole) [inline test/Middleware/RouteMiddlewareFactoryTest.php:69]: This is really not the best solution for this problem. But i think it's a little bit smarter than a reflection

File: composer.json [hunk 2]
```diff
@@ -28,18 +28,18 @@
         }
     },
     "require": {
-        "php": "^7.1",
+        "php": "^7.3 || ^8.0",
```
Comment (boesing) [inline composer.json]: Please use `~8.0.0` as we (TSC) decided to use this approach of allowing PHP packages for most laminas packages to guarantee that our packages are fully compatible with the PHP version when you run `composer update`

File: src/Middleware/ImplicitHeadMiddleware.php [hunk 3]
```diff
@@ -18,6 +18,7 @@
 use Psr\Http\Message\StreamInterface;
 use Psr\Http\Server\MiddlewareInterface;
 use Psr\Http\Server\RequestHandlerInterface;
+use zend\expressive\router\routeresult as ZendExpressiveRouteResult;
```
Comment (boesing) [inline src/Middleware/ImplicitHeadMiddleware.php]: Please make sure that the case matches the real namespace.
```suggestion
use Zend\Expressive\Router\RouteResult as ZendExpressiveRouteResult;
```

File: src/Middleware/RouteMiddleware.php [hunk 4]
```diff
@@ -16,6 +16,7 @@
 use Psr\Http\Message\ServerRequestInterface;
 use Psr\Http\Server\MiddlewareInterface;
 use Psr\Http\Server\RequestHandlerInterface;
+use zend\expressive\router\routeresult as ZendExpressiveRouteResult;
```
Comment (boesing) [inline src/Middleware/RouteMiddleware.php]:
```suggestion
use Zend\Expressive\Router\RouteResult as ZendExpressiveRouteResult;
```

File: test/Middleware/ImplicitHeadMiddlewareTest.php [hunk 5]
```diff
@@ -22,6 +22,7 @@
 use Psr\Http\Message\ServerRequestInterface;
 use Psr\Http\Message\StreamInterface;
 use Psr\Http\Server\RequestHandlerInterface;
+use zend\expressive\router\routeresult as ZendExpressiveRouteResult;
```
Comment (boesing) [inline test/Middleware/ImplicitHeadMiddlewareTest.php]:
```suggestion
use Zend\Expressive\Router\RouteResult as ZendExpressiveRouteResult;
```

**Review by boesing — CHANGES_REQUESTED** [review]

File: test/Middleware/RouteMiddlewareFactoryTest.php (hunk 1 above)
Comment (boesing) [reply test/Middleware/RouteMiddlewareFactoryTest.php:69]: Thats fine until the test is green ;-)

**Comment by boesing** [comment]
This is no BC break, so no need to mark the PR as such. 👍

**Review by boesing — APPROVED** [review]
LGTM

**Pull request merged**